File: src/experiment/evaluation-results/tinycua/experiment-4/workdir/frontend/src/schemas/blocks.ts
// Schema types for Block API responses and requests
export interface Block {
  id: string;
  type: 'text' | 'code' | 'heading' | 'image' | 'bullet-list';
  text_content?: string;
  parent_id?: string;
}

export interface BlockCreate extends Omit<Block, 'id'> {
  content?: string; // For text blocks
}

export interface BlockUpdate extends Partial<Omit<Block, 'id'>> {
  title?: string; // For heading blocks
}

export interface BlockResponse extends Omit<Block, 'id'> {
  id: string;
  type: 'text' | 'code' | 'heading' | 'image' | 'bullet-list';
  text_content?: string;
}

// Search response types
export interface SearchResultItem {
  type: 'block' | 'page' | 'property';
  id: string;
  title: string;
  content?: string | null;
  match_field?: string;
}

export interface SearchResponse {
  query: string;
  page: number;
  page_size: number;
  total_results: number;
  total_pages: number;
  results: SearchResultItem[];
}

File: src/experiment/evaluation-results/tinycua/experiment-4/workdir/src/schemas/blocks.py
"""Pydantic schemas for blocks API endpoints."""

from datetime import datetime
from typing import Optional, List
from uuid import uuid4

from sqlmodel import SQLModel, Field


class BlockCreate(SQLModel):
    """Schema for creating a new block."""
    id: Optional[str] = Field(default=None)  # Allow optional client-provided ID
    type: str = Field(default="text")  # text, code, heading, image, bullet-list
    content: Optional[str] = None  # For text blocks
    parent_id: Optional[str] = None


class BlockUpdate(SQLModel):
    """Schema for updating an existing block."""
    type: Optional[str] = Field(default=None)
    content: Optional[str] = Field(default=None)
    title: Optional[str] = Field(default=None)  # For heading blocks


class BlockResponse(SQLModel):
    """Schema for block response including full details."""
    id: str
    parent_id: Optional[str] = None
    type: str
    content: Optional[str] = None
    
    class Config:
        from_attributes = True


__all__ = ["BlockCreate", "BlockUpdate", "BlockResponse"]
